packages/authentication/test/acceptance/basic-auth.ts

```diff
@@ -91,8 +99,9 @@ describe('Basic Authentication', () => {
     });
   }
 
-  function givenAnApplication() {
-    app = new Application({
+  function givenAServer() {
+    app = new Application();
+    server = new RestServer(app, {
       components: [AuthenticationComponent],
```

Comment:
IIUC, this is configuring `AuthenticationComponent` for the REST server only, not for the entire app.

In one of my earlier comments, I was advocating for allowing users to configure components in a single place (app setup), so that they don't have to repeat component setup for all servers.

What is the current status/proposed design?

If we think authentication component should be transport specific (because the underlying implementation of sequence actions definitely will be transport specific), then I think we need to update our reference implementation accordingly. For example, rename `AuthenticationComponent` to `RestAuthenticationComponent`.